src/protocol/protocol.js

```diff
@@ -0,0 +1,263 @@
+const BN = require("bn.js");
+const SuperTokenModel = require("./../database/models/superTokenModel");
+
+class Protocol {
+
+    constructor(app) {
+        this.app = app;
+    }
+
+    async getAccountRealtimeBalance(token, address, timestamp) {
+        try {
+            this.app.client.addTotalRequest();
+            if (timestamp === undefined) {
+                timestamp = Math.floor(new Date().getTime() / 1000);
+            }
+
+            return this.app.client.superTokens[token.toLowerCase()].methods.realtimeBalanceOf(
+                address,
+                timestamp
+            ).call();
+        } catch (err) {
+            console.error(err)
+            throw Error(`account balance (${token}): ${err}`)
+        }
+    }
+
+    async getAccountRealtimeBalanceOfNow(token, address) {
+        try {
+            this.app.client.addTotalRequest();
+            return this.app.client.superTokens[token.toLowerCase()].methods.realtimeBalanceOfNow(
+                address
+            ).call();
+        } catch (err) {
+            console.error(err)
+            throw Error(`account balance (${token}): ${err}`)
+        }
+    }
+
+    async getAccountAgreementRealtimeBalance(token, account, timestamp) {
```

Comment:
method name and impl seem to be contradictory: name is generic (agreement), yet checks CFA agreement only